### PR TITLE
fix: avoid doubling cargo args in runnables

### LINKED ISSUE #17402: cargo runnables no longer work
It seems that after this PR, the tests have stopped running.

```
 *  Executing task: cargo cargo test --package sqruff-lib --lib -- rules::layout::LT13::tests::test_fail_leading_whitespace_comment --exact --show-output 

error: no such command: `cargo`

        View all installed commands with `cargo --list`
        Find a package to install `cargo` with `cargo search cargo-cargo`
```

_Originally posted by @gvozdvmozgu in https://github.com/rust-lang/rust-analyzer/issues/16840#issuecomment-2162227412_

### PR DESCRIPTION
**Pull request by davidbarsky**
Fixes #17402.

Sorry about this—I think I missed up the rebase!

### CODE REVIEW

File: editors/code/src/tasks.ts [hunk 1]
```diff
@@ -145,11 +145,13 @@ async function cargoToExecution(
         const cargoPath = await toolchain.cargoPath();
         const cargoCommand = definition.overrideCargo?.split(" ") ?? [cargoPath];
 
-        const args = [definition.command].concat(definition.args ?? []);
-        const fullCommand = [...cargoCommand, ...args];
-        const processName = unwrapUndefinable(fullCommand[0]);
+        const args = expectNotUndefined(
+            definition.args,
```
Comment (lnicola) [inline editors/code/src/tasks.ts:149]: I'm not really familiar with the code here, but shouldn't we use `definition.command` just like before?

Comment (davidbarsky) [reply editors/code/src/tasks.ts:149]: i'm not sure what you mean—can you expand?

Comment (lnicola) [reply editors/code/src/tasks.ts:149]: We're not using `definition.command` in the `cargo` case. I guess it's also present in `definition.args`, then?

Comment (Veykril) [reply editors/code/src/tasks.ts:149]: Shouldn't we drop the entire `cargoCommand` querying here? If I look at the previous PR, `createTask` already handles this now no?

Comment (davidbarsky) [reply editors/code/src/tasks.ts:149]: uh... yes. i think we can drop it.

**Pull request merged**